packages/core/src/render3/instructions/shared.ts

```diff
@@ -501,16 +502,23 @@ export function renderComponentOrTemplate<T>(
 function executeTemplate<T>(
     tView: TView, lView: LView, templateFn: ComponentTemplate<T>, rf: RenderFlags, context: T) {
   const prevSelectedIndex = getSelectedIndex();
+  const update = rf & RenderFlags.Update;
```

Comment:
May be rename this to indicate what the const represents?
```suggestion
  const isUpdatePhase = rf & RenderFlags.Update;
```